src/EnergyPlus/FileSystem.hh

```diff
@@ -49,43 +49,108 @@
 #define FileSystem_hh_INCLUDED
 
 #include <algorithm>
+#include <string>
+#include <fmt/format.h>
+#if __has_include(<filesystem>)
```

Comment:
`__has_include` might not necessarily work 100% how we want it to - many compiler vendors provide the headers for features they don't yet support, then give you a compile time warning if you try to use features from that header.

A better option is compiler feature test macros https://en.cppreference.com/w/User:D41D8CD98F/feature_testing_macros#C.2B.2B17 but those are not officially supported until C++20, and MSVC doesn't enable them unless you use C++20 mode (other compilers support them before C++20).

TL;DR, don't change anything, this is probably your best option for today, but I thought it was worth mentioning in case we do have an issue in the future with this.

Reply:
I think cmake's check_include_file might be the best option https://cmake.org/cmake/help/latest/module/CheckIncludeFile.html